staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go

```diff
@@ -249,6 +249,8 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope *RequestScope, forceWatc
 			}
 			klog.V(3).Infof("Starting watch for %s, rv=%s labels=%s fields=%s timeout=%s", req.URL.Path, opts.ResourceVersion, opts.LabelSelector, opts.FieldSelector, timeout)
 
+			ctx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
do we want the etcd timeout exactly the same as the rest request or slightly longer so that in the normal case, the rest timeout handler fires first?

Reply:
I was thinking about that and I think that doesn't really matter. In the code, both timeout and closed channel from the storage layer is treated the same way:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go#L227
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go#L231

So just for simplicity of thinking, I think that having them equal is easier. WDYT?

Reply:
If they're treated the same, then I agree it doesn't matter if they're set the same.